cpp/src/fil/fil.cu

```diff
@@ -328,7 +331,7 @@ struct forest {
   int fixed_block_count_ = 0;
   int max_shm_           = 0;
   // Optionally used
-  rmm::device_uvector<float> vector_leaf_;
+  rmm::device_uvector<std::uint8_t> vector_leaf_;
```

Comment:
If this is optionally used, can we consider `std::optional` here?

Reply:
If it's not used, it simply has the size of 0, so no need for `std::optional`. 

I've updated the documentation comment to make it clearer.